src/Raven.Server/Documents/Handlers/IndexHandler.cs

```diff
@@ -638,14 +639,15 @@ public async Task SetLockMode()
         [RavenAction("/databases/*/indexes/set-priority", "POST", AuthorizationStatus.ValidUser)]
         public async Task SetPriority()
         {
+            var guid = GetRaftRequestIdFromQuery();
             using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
             {
                 var json = await context.ReadForMemoryAsync(RequestBodyStream(), "index/set-priority");
                 var parameters = JsonDeserializationServer.Parameters.SetIndexPriorityParameters(json);
 
                 foreach (var name in parameters.IndexNames)
                 {
-                    await Database.IndexStore.SetPriority(name, parameters.Priority);
+                    await Database.IndexStore.SetPriority(name, parameters.Priority, $"{guid}/{name}");
```

Comment:
Are they guaranteed to be unique? Also elsewhere where you do this

Reply:
guid is unique.. as long as the name of the index is also unique (or any other suffix for the particular operation), it should be fine.

Reply:
I don't think we _require_ that the name will be unique, though.
Better to add an index number there too

Reply:
Would it make sense or even possible to pass 2 indexes with the same name in the same operation?

Reply:
Sense, no? 
Possible...